lib/orm.php

```diff
@@ -2541,4 +2589,124 @@ public function __unset( $key ) {
 	public function __isset( $key ) {
 		return $this->offsetExists( $key );
 	}
+
+	/**
+	 * Whether this site has persistent caching implemented.
+	 *
+	 * @return bool
+	 */
+	private static function has_persistent_cache() {
+		if ( self::$has_persistent_cache === false || self::$has_persistent_cache === true ) {
+			return self::$has_persistent_cache;
+		}
+
+		// Check if the site is using an external object cache.
+		if ( \function_exists( 'wp_using_ext_object_cache' )
+			&& \wp_using_ext_object_cache()
+		) {
+			self::$has_persistent_cache = true;
+			return true;
+		}
+
+		// Defensive coding - cover all bases.
+		// This will prevent errors below if WP_CONTENT_DIR is not defined.
+		if ( ! defined( 'WP_CONTENT_DIR' ) ) {
+			return false;
+		}
+
+		// Check if the site is using a persistent drop-in (object-cache.php).
+		// This _should_ be covered by the `wp_using_ext_object_cache()` check,
+		// but sometimes isn't depending on the timing of the cache.
+		if ( \file_exists( WP_CONTENT_DIR . '/object-cache.php' ) ) {
+			self::$has_persistent_cache = true;
+			return true;
+		}
```

Comment:
The WordPress performance plugin installs a `object-cache.php` for testing but object caching is not enabled.

Reply:
Yeap, good point!